userspace/libsinsp/user.cpp

```diff
@@ -271,89 +275,91 @@ scap_groupinfo *sinsp_usergroup_manager::groupinfo_map_insert(
 	return &grp;
 }
 
-scap_userinfo *sinsp_usergroup_manager::add_user(const string &container_id, uint32_t uid, uint32_t gid, const char *name, const char *home, const char *shell, bool notify)
+scap_userinfo *sinsp_usergroup_manager::add_user(const string &container_id, int64_t pid, uint32_t uid, uint32_t gid, const char *name, const char *home, const char *shell, bool notify)
 {
-	g_logger.format(sinsp_logger::SEV_DEBUG,
-			"adding user: container: %s, name: %s",
-			container_id.c_str(), name);
-
 	if (!m_import_users)
 	{
-		return nullptr;
+		static scap_userinfo fallback_user;
+		fallback_user.uid = uid;
+		fallback_user.gid = gid;
+		strlcpy(fallback_user.name, (uid == 0) ? "root" : "<NA>", sizeof(fallback_user.name));
+		strlcpy(fallback_user.homedir, (uid == 0) ? "/root" : "<NA>", sizeof(fallback_user.homedir));
+		strlcpy(fallback_user.shell, "<NA>", sizeof(fallback_user.shell));
+		return &fallback_user;
 	}
 
-	scap_userinfo *usr = get_user(container_id, uid);
-	if (!usr)
+	scap_userinfo *usr = get_user(HOST_CONTAINER_ID, uid);
+	if(usr)
 	{
-		bool inserted{false};
+		// Update user if it was already there
 		if (name)
 		{
-			usr = userinfo_map_insert(
-				m_userlist[container_id],
-				uid,
-				gid,
-				name,
-				home,
-				shell);
-			inserted = true;
-		}
-		else if (container_id.empty())
-		{
-#ifdef HAVE_PWD_H
-			// On Host, try to load info from db
-			auto* p = __getpwuid(uid, m_host_root);
-			if (p)
-			{
-				usr = userinfo_map_insert(
-					m_userlist[container_id],
-					p->pw_uid,
-					p->pw_gid,
-					p->pw_name,
-					p->pw_dir,
-					p->pw_shell);
-				inserted = true;
-			}
-#endif
+			strlcpy(usr->name, name, MAX_CREDENTIALS_STR_LEN);
+			strlcpy(usr->homedir, home, SCAP_MAX_PATH_SIZE);
+			strlcpy(usr->shell, shell, SCAP_MAX_PATH_SIZE);
 		}
+		return usr;
+	}
+
+	if (container_id.empty())
```

Comment:
clang-format would do this:
```suggestion
	if(container_id.empty())
```